Automattic/jetpack
# Iteration 7 Podcast Player

Remaining work on Podcast Player Block

#### Changes proposed in this Pull Request:
* Use CSS mask to color ME.js controls (#15257)
* Introduce CSS vars for colors (#15321)
* Enable mediaelement player color customization (#15332)
* Simplify creating color change handler (#15335)
* Memoize colors object (#15336)
* Theme Compatibility Fixes (#15320)
* Lint everything (#15281)
* Fix Sound Icon Position (#15343)
* Remove bottom border on podcast header (#15355)
* Decode all HTML entities (#15351) 
* Consistent gutter spacing (#15356)
* Fix Sound Icon Position... including IE (#15371)
* Fix alignment issues in placeholder (#15376)
* Apply color customization to error state (#15394)
* Add an error boundary (#15369)
* Break components into their own files (#15442)
* Make speaker icons consistent (#15445) 
* Instruct IE11 user on how to activate audio support (#15451)
* Add external embed detection (#15446)
* Fix error message alignment (#15462)
* Make current track title inherit primary color (#15463)
* Center track icon vertically (#15465)

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Polishes Podcast Player block after CfT

#### Testing instructions:

_**NOTE: it should be tested in Simple sites**_

General URL for testing: https://anchor.fm/s/9400d7c/podcast/rss

* To test customization of controls colors (in modern browsers), go to the bottom of [`styles.css`](https://github.com/Automattic/jetpack/compare/feature/podcast-player...try/podcast-player-mejs-svg-colors?expand=1#diff-23936c23c59259d544403d0cc941ea1d) and set any color to the [background-color property of the ME.js button](https://github.com/Automattic/jetpack/compare/feature/podcast-player...try/podcast-player-mejs-svg-colors?expand=1#diff-23936c23c59259d544403d0cc941ea1dR340). Refresh the page and see if the color has been set. Play with the controls by clicking them - it should toggle colored `play`/`pause`, `mute`/`unmute` and `replay` SVGs.
* Play a track in the editor: The sound icon should look centered with the first line of the track
* Reduce the window size so the track titles wrap. The sound icon should still be centered with the first line of the track
* The sound icon being added should not increase the height of the track `<a>` element
* Test with one of these feeds and make sure there are no encoded html entities: https://whatsurfave.com/category/podcast/feed/, https://anchor.fm/s/9400d7c/podcast/rss
* Add the Podcast Player (Beta), Youtube embed, and RSS blocks. Compare the three blocks and check that input and button elements are aligned and that the blocks look consistent.
* Block a request / go offline to throw the episode error. In the editor, set primary/secondary/background colors and observe whether the error elements are now affected by color customization.
* Try adding `throw new Error( 'Test' )` somewhere in render functions of our components or inside functional components. For example, as the very first line in `render()` inside `audio-player.js`. Inside the editor, the player should be replaced by a simple error message. On the frontend, we have an additional handling that suppresses the error and instead unmounts the React component and brings back the static markup.
* In IE 11: Test both with the "audio support for webpages" setting unchecked and checked. Observe whether what you see reflects the states displayed in the #15451 issue description.
* Test with multiple URLs and confirm they all embed correctly. Some examples:
  - http://example.com - should show error
  - https://distributed.blog/category/podcast/feed/ - direct RSS link, should use our player
  - https://anchor.fm/startapodcast/episodes/Whats-your-podcast-about-e17krq/a-a2q3ft - regular webpage with `<link>` to RSS, should also use our player
  - https://open.spotify.com/show/7gozmLqbcbr6PScMjc0Zl4, invalid RSS but valid oEmbed, should replace Podcast Player with the spotify block
  - https://www.youtube.com/watch?v=dQw4w9WgXcQ, again invalid as RSS but works as oEmbed, should replace our block with youtube block


#### Proposed changelog entry for your changes:
* Not sure if needed at this time?

#### Follow-up issues
https://github.com/Automattic/jetpack/issues/15443, https://github.com/Automattic/jetpack/issues/15444, #15459
